nav2_system_tests/src/localization/test_localization_node.cpp

```diff
@@ -53,7 +54,7 @@ class TestAmclPose : public ::testing::Test
     subscription_ = node->create_subscription<geometry_msgs::msg::PoseWithCovarianceStamped>(
       "amcl_pose", rclcpp::QoS(rclcpp::KeepLast(1)).transient_local().reliable(),
       std::bind(&TestAmclPose::amcl_pose_callback, this, _1));
-    initial_pose_pub_->publish(testPose_);
+    initial_pose_pub_->publish(*testPose_);
```

Comment:
isnt't it `initial_pose_pub_->publish(std::move(*testPose_))`?

Reply:
Not quite, you're moving the pointer ownership, not the object.